client/src/components/home/HomePage.tsx

```diff
@@ -159,19 +165,28 @@ export default class HomePage extends Component<IHomePageProps, IHomePageState>
         <Row />
         <Row>
           <Col md={12}>
-            <FormGroup bsSize="large" controlId="address" validationState={this.getValidationState()}>
+            <FormGroup bsSize="large" controlId="address" validationState={this.getValidationState().validation}>
               <ControlLabel style={{ fontSize: "2em" }}>Search for the voting</ControlLabel>
               <HelpBlock>
-                Enter a valid keccak256 Ethereum address. Use all upper- or lowercase characters to ignore{" "}
-                <a href="https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md">checksum validation</a>.
+                Enter a valid keccak256 Ethereum address. It must start with '0x' and be exactly 42 characters long.
               </HelpBlock>
               <InputGroup>
-                <FormGroup bsSize="large" controlId="address" validationState={this.getValidationState()}>
-                  <FormControl type="text" value={this.state.searchBoxText} onChange={this.handleSearchBoxChange} />
+                <FormGroup bsSize="large" controlId="address">
+                  <FormControl
+                    type="text"
+                    value={this.state.searchBoxText}
+                    onChange={this.handleSearchBoxChange}
+                    placeholder="Enter voting adress"
+                    onKeyPress={(event) => {
+                      if (event.key === "Enter") {
+                        this.searchVoting();
+                      }
+                    }}
+                  />
                   <FormControl.Feedback />
                 </FormGroup>
                 <InputGroup.Button>
-                  <Button bsSize="large" onClick={this.searchVoting}>
+                  <Button bsSize="large" onClick={this.searchVoting} disabled={this.getValidationState().disableSearch}>
```

Comment:
If you make `#getValidationState` return `Validation` type, you can write
`disabled={this.getValidationState() === Validation.Error}`